iso/step_add_cdrom.go

```diff
@@ -38,13 +39,20 @@ func (s *StepAddCDRom) Run(_ context.Context, state multistep.StateBag) multiste
 			return multistep.ActionHalt
 		}
 	}
-
+  
 	ui.Say("Mount ISO images...")
-	for _, path := range s.Config.ISOPaths {
-		if err := vm.AddCdrom(s.Config.CdromType, path); err != nil {
+	if path, ok := state.GetOk(s.RemotePath); ok {
+		if err := vm.AddCdrom(path.(string)); err != nil {
 			state.Put("error", fmt.Errorf("error mounting an image: %v", err))
 			return multistep.ActionHalt
 		}
+	} else {
```

Comment:
we need to reverse the order. existing images are mounted first (like OS distro), dynamically uploaded image later